### PR TITLE
Added download button

### LINKED ISSUE #4: Add option to download create YAML
The form should offer two buttons, one to "show" the created YAML and one to download it as `<device_id>.yml`.

### PR DESCRIPTION
**Pull request by Aditi-1400**
- Added the download file button
- Removed react-select for now due to delayed state update 

Closes #4

### CODE REVIEW

File: form/src/App.js [hunk 1]
```diff
@@ -29,26 +39,26 @@ function App() {
...
+  //       classNamePrefix="select"
+  //       options={props.options.enumOptions}
+  //       placeholder={props.label}
+  //       isSearchable={true}
+  //       onChange={async (e) => {
+  //         await props.onChange(e.value);
+  //       }}
+  //     ></Select>
+  //   );
+  // };
```
Comment (aparcar) [inline form/src/App.js]: Please see your other PR on how to deal with this. 1. have it in a different commit/PR and second don't *comment* code but remove it, it stays in the git log anyway.

File: form/src/App.js [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
+  const [isOpen, setIsOpen] = React.useState(false); 
   const [content, setContent] = React.useState("Transitioning...");
   const [formData, setFormData] = React.useState({});
+  const [url, setUrl] = React.useState("");
+  const [filename, setFilename] = React.useState("");
+
+  useEffect(() => {
+    var data = new Blob([content]);
+    setUrl(URL.createObjectURL(data));
+    const name = formData.vendor+formData.model+formData.variant+".yaml"
```
Comment (aparcar) [inline form/src/App.js]: This is mostly fine, however please modify the name to follow the `device_id`. It's defined by the following pseudo code:

```
filename = "${ vendor.toLowerCase() }_${ model.toLowerCase()"
if (variant != "") {
  filename = filename + '-' + variant.toLowerCase()
}
filename = filename.replace(' ', '-')
filename = filename.replaceNonAsciiChars()
```

So, all lowercase, no spaces, only ASCII chars and attach the `variant` only if defined. Currently the filename would become something like `TP-Link Router 5000undefined.yml` while it should be `tp-link_router-5000.yaml` and `Foobar Test 4000 v3` should become `foobar_test-4000-v3`.

**Review by aparcar — CHANGES_REQUESTED** [review]

**Comment by aparcar** [comment]
Please rebase this PR and implement the filename change I suggested.

**Comment by aparcar** [comment]
Thanks for the update! The `variant` handling is still incomplete but looks good apart from that:
image

**Comment by aparcar** [comment]
This commit contains a merge commit which can be removed via rebasing. I used the following commands to do so, please try locally and update the PR:

```
git rebase main # this gives an error message of things not lining up
git diff # check out what's actually wrong
vim form/src/App.js # here is the issue, use whatever editor you like, fix the file
git add form/src/App.js
git rebase --continue
git show # just to verify everything is in shape
```

Let me know if there are issues

**Comment by Aditi-1400** [comment]
I've made the changes, Let me know if there are any other issues!

File: form/src/App.js [hunk 3]
```diff
@@ -13,6 +13,23 @@ function App() {
...
+  const [filename, setFilename] = React.useState("");
+
+  useEffect(() => {
+    var data = new Blob([content]);
+    setUrl(URL.createObjectURL(data));
+    var {vendor, model, variant} = formData;
+    if((vendor !== undefined && model !== undefined) || variant !== undefined){
+    var name = vendor.toLowerCase()+'_'+ model.toLowerCase();
+    if(variant !== undefined){
+      name = name +'-'+ variant + ".yaml";
```
Comment (aparcar) [inline form/src/App.js]: the `yaml` suffix is only added if a variant is set, it should be added in any case.

**Comment by aparcar** [comment]
Excellent, thanks!

**Pull request merged**